components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2627,6 +2628,36 @@ where
         self.fsm.hibernate_state.count_vote(from.get_id());
     }
 
+    fn on_trace_peer_availability_info(&mut self, from: &metapb::Peer, msg: &ExtraMessage) {
+        if !self.fsm.peer.is_leader() {
```

Comment:
What if it sent out the message when it was still leader, but got response after became follower? Then it will step into dead loop.

Reply:
How about split `MsgTracePeerAvailabilityInfo` into a request type and a response type? When a pre-leader peer got a response type, it drops the msg.

Reply:
It makes more sense.